docs/DeveloperGuide.md

```diff
@@ -588,15 +605,41 @@ Similar to [<ins>UC04](#use-case-uc04-create-a-contact) except adding a task to
 
       Use case resumes at step 1.
 
-### **Use case: UC08 — Add Address to Contact** 
-Similar to [<ins>UC07](#use-case-uc07-add-phone-number-to-contact) except duplicated addresses are allowed
+### **Use case: UC10 — Add Address to Contact** 
+Similar to [<ins>UC09](#use-case-uc09-add-phone-number-to-contact) except duplicated addresses are allowed
+
+### **Use case: UC11 — Add Email to Contact**
+Similar to [<ins>UC09](#use-case-uc09-add-phone-number-to-contact) except duplicated email addresses are allowed
+
+### **Use case: UC12 — Tag a Contact**
 
-### **Use case: UC09 — Add Email to Contact**
-Similar to [<ins>UC07](#use-case-uc07-add-phone-number-to-contact) except duplicated email addresses are allowed
+**MSS**
+
+1. User <ins>lists all tags [(UC04)](#use-case-uc04-list-all-tags)</ins>.
+2. User requests to tag a contact with a specified tag.
+3. The system displays the updated contact with the new tag assigned to them, in the list of contacts.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The contact list is empty.
+  Use case ends.
+
+* 1b. The tag list is empty.
+  Use case ends.
+
+* 2a. The provided tag name does not exist in the System.
```

Comment:
Should be nontechnical so maybe don't need to mention system, can change to: 'The provided tag name does not exist"?

Reply:
sure, i'll change it